lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,6 +109,40 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not (isinstance(word_frequencies, dict) or word_frequencies is None) or not
+        isinstance(num_merges, int)):
+        return None
+    token_pairs = count_tokens_pairs(word_frequencies)
+    while (num_merges > 0) and len(token_pairs) > 0:
+        num_merges -= 1
+        max_freq = 0
+        max_pairs = []
+        for freq in token_pairs.values():
```

Comment:
it looks like you are re-implementing the built-in function `max`,  or at least method `sorted`. Consider passing the necessary features via `key` argument.